scripts/fix_demux_artifact_names.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+DESC = """
+This script addresses a specific LIMS bug in which the demux artifacts of a
+demultiplexing step are associated to multiple samples, instead of a single sample.
+
+The bug is described in deviation #312 on AM.
+
+I.e. a list of demux artifacts for a pool which should be e.g.
+    P34304_1004 (FASTQ reads)
+    P34304_1005 (FASTQ reads)
+    P34304_1006 (FASTQ reads)
+instead becomes e.g.
+    P34304_1004 + P34304_1005 + P34304_1006 (FASTQ reads)
+    P34304_1005 + P34304_1006 (FASTQ reads)
+    P34304_1005 + P34304_1006 (FASTQ reads)
+
+The identity of each demux artifact is dubious, but they do appear to maintain
+the correct reagent label.
+
+This script
+ - Iterates through each pool in the process
+    - Creates an SQL-derived sample-label mapping for the pool contents
+    - Iterates through each demux artifact associated with the pool
+        - Uses the sample-label mapping to find the "correct" sample name
+        - Renames the demux artifact to the "correct" sample name
+
+The demux artifacts will still have multiple samples associated with them, but at least
+the name will be corrected to facilitate downstream processing.
+
+"""
+import logging
+from argparse import ArgumentParser, Namespace
+from datetime import datetime as dt
+
+from genologics.config import BASEURI, PASSWORD, USERNAME
+from genologics.entities import Process
+from genologics.lims import Lims
+
+from scripts.generate_minknow_samplesheet import get_pool_sample_label_mapping
```

Comment:
Might be worth moving this function where it's more obvious it's used for Illumina as well?

Reply:
3e8c4ccf71f1e5e9d0adcc073a0cb3dc084e0886